server/bound_queries.h

```diff
@@ -143,7 +143,7 @@ struct BoundGroupByQuery {
 
     const Dataset & from;
     WhenExpression when;
-    std::shared_ptr<SqlExpression> where;
+    const SqlExpression & where;
```

Comment:
My concern was that the owner would free that object before BoundGroupByQuery is done with it.